src/image/PhotoCalib.cc

```diff
@@ -0,0 +1,409 @@
+                                   bool throwOnNegativeFlux) const
+{
+    countsToMagnitudeArray(sourceCatalog, countsField, result, throwOnNegativeFlux);
+}
+
+void PhotoCalib::countsToMagnitude(afw::table::SourceCatalog &sourceCatalog,
+                       const std::string &countsField,
+                       const std::string &outField,
+                       bool throwOnNegativeFlux) const
+{
+    throw LSST_EXCEPT(pex::exceptions::LogicError,
+                      "Will not be implemented until RFC-322 is implemented. See DM-10155.");
+}
+
+// ------------------- other utility methods -------------------
+
+double PhotoCalib::magnitudeToCounts(double magnitude) const
+{
+    return pow(10, magnitude/-2.5)*_countsMag0;
+}
+
+std::shared_ptr<math::BoundedField> PhotoCalib::computeScaledZeroPoint() const
+{
+    return *(_zeroPoint)/_countsMag0;
+}
+
+std::shared_ptr<math::BoundedField> PhotoCalib::computeScalingTo(std::shared_ptr<PhotoCalib> other) const
+{
+    throw LSST_EXCEPT(pex::exceptions::LogicError, "Not Implemented: See DM-10154.");
+}
+
+bool PhotoCalib::operator==(PhotoCalib const& rhs) const
+{
+    return (_countsMag0 == rhs._countsMag0 && _countsMag0Sigma == rhs._countsMag0Sigma &&
+        (*_zeroPoint) == *(rhs._zeroPoint));
+}
+
+double PhotoCalib::computeCountsMag0(std::shared_ptr<afw::math::BoundedField> zeroPoint) const
+{
+    return zeroPoint->mean();
+}
+
+std::ostream & operator<<(std::ostream & os, PhotoCalib const & photoCalib) {
+    os << *(photoCalib._zeroPoint) << " with ";
+    return os << "mean: " << photoCalib._countsMag0 << " sigma: " << photoCalib._countsMag0Sigma;
+}
+
+// ------------------- persistence -------------------
+
+namespace {
+
+int const PHOTOCALIB_TABLE_CURRENT_VERSION = 1;
+
+class PersistenceHelper {
+public:
+    table::Schema schema;
+    table::Key<double> countsMag0;
+    table::Key<double> countsMag0Sigma;
+    table::Key<table::Flag> isConstant;
+    table::Key<int> field;
+
+    // No copying
+    PersistenceHelper (const PersistenceHelper&) = delete;
+    PersistenceHelper& operator=(const PersistenceHelper&) = delete;
+    // No moving
+    PersistenceHelper (PersistenceHelper&&) = delete;
+    PersistenceHelper& operator=(PersistenceHelper&&) = delete;
+
+    static PersistenceHelper const & get() {
+        static PersistenceHelper const instance;
+        return instance;
+    }
+
+    PersistenceHelper(int tableVersion=PHOTOCALIB_TABLE_CURRENT_VERSION) :
+    schema(), countsMag0(), countsMag0Sigma(), field() {
+        countsMag0 = schema.addField<double>("countsMag0", "counts of a zero-magnitude object", "count");
+        countsMag0Sigma = schema.addField<double>("countsMag0Sigma", "1-sigma error on countsmag0", "count");
+        isConstant = schema.addField<table::Flag>("isConstant", "Is this spatially-constant?");
+        field = schema.addField<int>("field", "archive ID of the BoundedField object");
+    }
+
+    PersistenceHelper(table::Schema const & s) :
```

Comment:
@jdswinbank it's published at https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/afw_table_persistence.html. Maybe a cross-reference needs to be added from the persistence classes?